pkg/deviceplugin/manager.go

```diff
@@ -40,20 +64,26 @@ func (manager *NicManager) GetDeviceFiles() []*pluginapi.DeviceSpec {
 }
 
 // NewNicManager allocates and initialises a new NicManager
-func NewNicManager(maxPods int, needNic bool) (*NicManager, error) {
+func NewNicManager(
+	config NicManagerConfig,
+) (*NicManager, error) {
 	nics, err := queryNics()
 	if err != nil {
 		return nil, err
 	}
-	if len(nics) == 0 && needNic {
+	if len(nics) == 0 && config.NeedNic {
 		return nil, errors.New("no sfc devices found")
 	}
 	manager := &NicManager{
-		interfaces:     nics,
-		maxPodsPerNode: maxPods,
+		interfaces: nics,
+		config:     config,
 	}
 	manager.envs = make(map[string]string)
 	manager.initDevices()
+
+	if manager.config.Preload && manager.config.MountOnload {
```

Comment:
Yeah, we could probably do something with [kubebuilder validations](https://book.kubebuilder.io/reference/markers/crd-validation), but since in theory the device plugin should work without the controller we would still need a check in the device plugin.